### PR TITLE
IBX-2752 Raptor CDP connector

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [IBX-2752](https://issues.ibexa.co/browse/IBX-2752)
| Versions      | 4.2 and 3.3

- 'php bin/console ibexa:cdp:stream-user-data' command options will be added in a separate PR
- the list of tracking events will be added separately
- links to Raptor will be removed and replaced by the Ibexa doc on a later stage

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/cdp/cdp.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Customer Data Platform (CDP) [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]
+
+## What is Ibexa CDP
+
+Ibexa CDP helps you solve one of the hardest challenges facing business world today:
+building unique experiences for your customers.
+With Ibexa CDP you will be able to track and aggregate data on your customers' activity on multiple channels.
```
Comment (juskora) [inline docs/cdp/cdp.md:7]:
```suggestion
With Ibexa CDP you will be able to track and aggregate data of your customers' activity on multiple channels.
```

File: docs/cdp/cdp.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## How it works
+
+Ibexa CDP unifies customer data across your organization
+to help you activate your users and provide them with real-time engagement.
+With defined audiences you can target your user segments at the right time,
+through the most used channel, with the relevant message, content, or products.
+
+The customer data are collected through the system of trackers embedded in different parts of your page.
+For more information on activation and trackers go to [CDP activation documentation](cdp_activation.md).
```
Comment (juskora) [inline docs/cdp/cdp.md:20]:
```suggestion
For more information on activation and trackers, see[CDP activation documentation](cdp_activation.md).
```

File: docs/cdp/cdp_activation.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
+# CDP activation  [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]
+
+## Configuration
+
+To configure Ibexa CDP you have to edit the `config/packages/ibexa_cdp.yaml` file:
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:5]:
```suggestion
To configure Ibexa CDP, edit the `config/packages/ibexa_cdp.yaml` file:
```

File: docs/cdp/cdp_activation.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+Fill in name and identifier for a segment group.
+Choose wisely, as once connected to CDP Segment Group cannot be changed.
+
+![Creating a new segment group](img/cdp_create_segment_group.png)
+
+Next, add a segment group identifier to the configuration.
+
+!!! caution "Ibexa CDP Segment Group"
+
+    Once you create the Segment Group in the Back Office and connect it to Ibexa CDP you cannot change it in any way, including edit its name.
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:38]:
```suggestion
    After you create the Segment Group in the Back Office and connect it to Ibexa CDP, you cannot change it in any way, including edit its name.
```

File: docs/cdp/cdp_activation.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+## Account number
+
+Now, fill in the account number.
+Log in to Ibexa CDP and select available accounts in the top right corner.
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:43]:
```suggestion
Log in to Ibexa CDP and in the top right corner, select available accounts.
```

File: docs/cdp/cdp_activation.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+![List of available accounts](img/cdp_accounts.png)
+
+A pop-up window will appear with a list of all available accounts and their numbers.
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:47]:
```suggestion
A pop-up window displays with a list of all available accounts and their numbers.
```

File: docs/cdp/cdp_activation.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+## User Data Streaming
+
+You need to specify a source of the user data that Ibexa CDP will connect to.
+To do so, go to **Data Manager** in **Tools** section and select **Create new dataflow**.
+It will take you to a Dataflow Creator, where in five steps you will set up a data streaming.
+
+### General Information
+
+In the **General Information** section specify dataflow name,
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:59]:
```suggestion
In the **General Information** section, specify dataflow name,
```

File: docs/cdp/cdp_activation.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+choose **Stream File** as a source of user data and **CDP** as a destination,
+where they will be sent for processing.
+
+### Download
+
+In the **Download** section select **Stream file**.
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:65]:
```suggestion
In the **Download** section, select **Stream file**. 
```

File: docs/cdp/cdp_activation.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+Next, you need to export your user data to the CDP.
+Got to your installation and use this command:
+
+```bash
+php bin/console ibexa:cdp:stream-user-data --draft
+```
+
+There are two versions of this command `--draft/--no-draft`.
+The first one is used to send the test user data to the Data Manager.
+If it passes a validation test in the **Activation** section, you will use the latter one to send a full version.
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:78]:
```suggestion
If it passes a validation test in the **Activation** section, use the latter one to send a full version.
```

File: docs/cdp/cdp_activation.md [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+
+Next, go back to Ibexa CDP and select **Validate & download**.
+If the file passes, you will see a confirmation message.
+Now, you can move on to the **File mapping** section.
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:82]:
```suggestion
Now, you can go to the **File mapping** section.
```

File: docs/cdp/cdp_activation.md [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+There, you can choose conditions from `did`, `did not` or `have`.
+The conditions `did` and `did not` allow you to use events like buy, visit or add to a basket from online tracking.
+- `have` conditions are tied to personal characteristics and can be used to track the sum of all buys or top-visited categories.
+
+In the Audience Builder you can also connect created audiences to the activations.
+
+## Activation
+
+Activation synchronises data from Ibexa CDP to the Ibexa DXP.
+When you specify a segment, you can Activate it on multiple communication channels, such as newsletters or commercials.
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:144]:
```suggestion
When you specify a segment, you can activate it on multiple communication channels, such as newsletters or commercials.
```

File: docs/cdp/cdp_activation.md [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+You can configure multiple activations based on one user data streaming.
+
+First, select **Activations** from the menu bar and create a new **Ibexa** activation.
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:147]:
```suggestion
First, from the menu bar, select **Activations**  and create a new **Ibexa** activation.
```

File: docs/cdp/cdp_activation.md [hunk 13]
```diff
@@ -0,0 +1,196 @@
...
+```
+
+For tracing to be effective, you also need to send ID of a logged-in user in the same way.
+Add the user ID information by using below script:
+
+```js
+raptor.push("setRuid","USER_ID_HERE")
+```
+
+For more information on tracking events follow [the documentation]()https://support.raptorsmartadvisor.com/hc/en-us/articles/201912411-Tracking-Events.
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:196]:
```suggestion
For more information on tracking events, see [the documentation]()https://support.raptorsmartadvisor.com/hc/en-us/articles/201912411-Tracking-Events.
```

File: docs/cdp/cdp_activation.md [hunk 14]
```diff
@@ -0,0 +1,196 @@
...
+where they will be sent for processing.
+
+### Download
+
+In the **Download** section select **Stream file**. 
+Copy generated steam ID and paste it into the `config/packages/ibexa_cdp.yaml` file under `stream_id`.
+It allows you to establish a datastream from the Streaming API into the Data Manager.
+
+Next, you need to export your user data to the CDP.
+Got to your installation and use this command:
```
Comment (juskora) [inline docs/cdp/cdp_activation.md:70]:
```suggestion
Go to your installation and use this command:
```

**Review by juskora — CHANGES_REQUESTED** [review]

**Review by juskora — APPROVED** [review]

File: docs/cdp/cdp.md [hunk 15]
```diff
@@ -0,0 +1,20 @@
+# Customer Data Platform (CDP) [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]
```
Comment (webhdx) [inline docs/cdp/cdp.md:1]: Not sure if I understand the badge correctly - if it means CDP is available from Experience edition then this is wrong as it's a standalone package that can be installed alongside every product edition (Content, Experience, Commerce). Maybe we should add Content badge as well.

File: docs/cdp/cdp.md [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+
+## How it works
+
+Ibexa CDP unifies customer data across your organization
+to help you activate your users and provide them with real-time engagement.
+With defined audiences you can target your user segments at the right time,
+through the most used channel, with the relevant message, content, or products.
+
+The customer data are collected through the system of trackers embedded in different parts of your page.
+For more information on activation and trackers, see[CDP activation documentation](cdp_activation.md).
```
Comment (webhdx) [inline docs/cdp/cdp.md:20]:
```suggestion
For more information on activation and trackers, see [CDP activation documentation](cdp_activation.md).
```

File: docs/cdp/cdp_activation.md [hunk 17]
```diff
@@ -0,0 +1,196 @@
+# CDP activation  [[% include 'snippets/experience_badge.md' %]] [[% include 'snippets/commerce_badge.md' %]]
```
Comment (webhdx) [inline docs/cdp/cdp_activation.md:1]: See my previous comment about badges.

**Review by webhdx — CHANGES_REQUESTED** [review]
I have some small remarks but nothing major.

File: docs/cdp/cdp.md (hunk 15 above)
Comment (juskora) [reply docs/cdp/cdp.md:1]: So let's remove them.

File: docs/cdp/cdp_activation.md [hunk 18]
```diff
@@ -0,0 +1,196 @@
...
+                account_number: 123456
+                user_data_streaming:
+                    stream_id: 00000000-00000000-00000000-00000000
+                activations:
+                    client_id: '%env(CDP_ACTIVATION_CLIENT_ID)%'
+                    client_secret: '%env(CDP_ACTIVATION_CLIENT_SECRET)%'
+                    segment_group_identifier: example_segment_group_identifier
+```
+
+All configuration settings are described bellow.
```
Comment (webhdx) [inline docs/cdp/cdp_activation.md:21]: A typo:
```suggestion
All configuration settings are described below.
```

**Review by webhdx — APPROVED** [review]

File: docs/cdp/cdp.md [hunk 19]
```diff
@@ -0,0 +1,20 @@
+# Customer Data Platform (CDP)
+
+## What is Ibexa CDP
+
+Ibexa CDP helps you solve one of the hardest challenges facing business world today:
+building unique experiences for your customers.
+With Ibexa CDP you will be able to track and aggregate data of your customers' activity on multiple channels.
+It will allow you to create individual customer profiles that enable you to personalize their experience on your platform.
+
+![](img/cdp_control_panel.png)
```
Comment (MagdalenaZuba) [inline docs/cdp/cdp.md:10]:
```suggestion
![Ibexa CDP control panel](img/cdp_control_panel.png)
```

File: docs/cdp/cdp_activation.md [hunk 20]
```diff
@@ -0,0 +1,196 @@
...
+```
+
+For tracing to be effective, you also need to send ID of a logged-in user in the same way.
+Add the user ID information by using below script:
+
+```js
+raptor.push("setRuid","USER_ID_HERE")
+```
+
+For more information on tracking events, see [the documentation]()https://support.raptorsmartadvisor.com/hc/en-us/articles/201912411-Tracking-Events.
```
Comment (MagdalenaZuba) [inline docs/cdp/cdp_activation.md:196]:
```suggestion
For more information on tracking events, see [the documentation](https://support.raptorsmartadvisor.com/hc/en-us/articles/201912411-Tracking-Events).
```

File: docs/cdp/cdp.md [hunk 21]
```diff
@@ -0,0 +1,20 @@
+# Customer Data Platform (CDP)
```
Comment (MagdalenaZuba) [inline docs/cdp/cdp.md:1]:
```suggestion
---
description: Ibexa CDP a software system designed to collect and organize customer data from multiple sources to build comprehensive customer profiles.
---

# Customer Data Platform (CDP)
```

File: docs/cdp/cdp_activation.md [hunk 22]
```diff
@@ -0,0 +1,196 @@
+# CDP activation
```
Comment (MagdalenaZuba) [inline docs/cdp/cdp_activation.md:1]:
```suggestion
---
description: Step by step activation procedure of Ibexa CDP.
---

# CDP activation
```

File: docs/cdp/cdp_installation.md [hunk 23]
```diff
@@ -0,0 +1,36 @@
+# Ibexa CDP installation
```
Comment (MagdalenaZuba) [inline docs/cdp/cdp_installation.md:1]:
```suggestion
---
description: Imnstallation od standalone Ibexa CDP package.
---

# Ibexa CDP installation
```

**Pull request merged**